gems/smithy-client/spec/smithy-client/waiters_spec.rb

```diff
@@ -0,0 +1,708 @@
+                end.to_not raise_error
+              end
+
+              it 'fails when output property does not match' do
+                client.stub_responses(:get_widget, { string_property: 'unexpected string' })
+                expect do
+                  client.wait_until(:output_string_property_matcher, input, max_wait_time: 0)
+                end.to raise_error(MaxWaitTimeExceededError)
+              end
+
+              it 'fails when output property is nil' do
+                client.stub_responses(:get_widget, {})
+                expect do
+                  client.wait_until(:output_string_property_matcher, input, max_wait_time: 0)
+                end.to raise_error(MaxWaitTimeExceededError)
+              end
+            end
+
+            context 'boolean equals comparator' do
+              let(:matchers) do
+                {
+                  'OutputBooleanPropertyMatcher' => {
+                    'acceptors' => [
+                      {
+                        'state' => 'success',
+                        'matcher' => {
+                          'output' => {
+                            'path' => 'booleanProperty',
+                            'expected' => 'false',
+                            'comparator' => 'booleanEquals'
+                          }
+                        }
+                      }
+                    ]
+                  }
+                }
+              end
+
+              it 'succeeds when output matches' do
+                client.stub_responses(:get_widget, { boolean_property: false })
+                expect do
+                  client.wait_until(:output_boolean_property_matcher, input, max_wait_time: 60)
+                end.to_not raise_error
+              end
+
+              it 'retries and succeeds when matched' do
+                client.stub_responses(
+                  :get_widget,
+                  { boolean_property: true },
+                  { boolean_property: true },
+                  { boolean_property: false }
+                )
+                expect_any_instance_of(Waiter).to receive(:delay).twice.and_call_original
+                expect do
+                  client.wait_until(:output_boolean_property_matcher, input, max_wait_time: 60)
+                end.to_not raise_error
+              end
+
+              it 'fails when output property does not match' do
+                client.stub_responses(:get_widget, { boolean_property: true })
+                expect do
+                  client.wait_until(:output_boolean_property_matcher, input, max_wait_time: 0)
+                end.to raise_error(MaxWaitTimeExceededError)
+              end
+
+              it 'fails when output property is nil' do
+                client.stub_responses(:get_widget, {})
+                expect do
+                  client.wait_until(:output_boolean_property_matcher, input, max_wait_time: 0)
```

Comment:
Might be useful to DRY this up a little. You could either, extract options into a let() variable, always set max_wait_time to 60, and just pass it into all these for convenience.

Or make a method like:
```
def wait(waiter_name)
  client.wait_until(waiter_name, { string_property: 'input' }, { max_wait_time: 60 })
end

expect { wait(:output_boolean_property_matcher) }
  .to raise_error(MaxWaitTimeExceededError)
```